src/DirectoryTree.py

```diff
@@ -0,0 +1,111 @@
+import os, sys, shutil
+from os.path import expanduser
+from File import File
+
+class DirectoryTree:
+    def __init__(self, path):
+        self.path = expanduser(path) if path.startswith(("~", "/")) else path
+        self.root, self.files = self.create_tree_with_computed_sizes(self.path)
+
+    def create_tree_with_computed_sizes(self, path):
+        files_dict = {}
+        for dirname, subdirs, files in os.walk(path, topdown=False):
+            root = File(dirname)
+            dir_size = 0
+            for entry in files:
+                next_path = os.path.abspath(os.path.join(dirname, entry))
+                if self.file_to_skip(next_path):
+                    continue
+                next_file = File(next_path)
+                child_size = os.path.getsize(next_path)
+                next_file.set_size(child_size)
+                dir_size += child_size
+                root.add_next_child(next_file)
+                files_dict[next_path] = next_file
+
+            for entry in subdirs:
+                next_path = os.path.abspath(os.path.join(dirname, entry))
+                if self.file_to_skip(next_path):
+                    continue
+                next_file = files_dict[next_path]
+                dir_size += next_file.get_size()
+                root.add_next_child(next_file)
+
+            root.set_size(dir_size)
+            for child in root.get_children():
+                child.set_parent_size(dir_size)
+            files_dict[dirname] = root
+        
+        files_dict[path].set_parent_size(0)
+        return files_dict[path], files_dict
+
+    def check_access(self, path):
+        return os.access(path, os.R_OK)
+
+    def file_to_skip(self, path):
```

Comment:
better name would be something like `should_skip_file`, `file_to_skip` suggests that the function returns a file to skip o.o?

Reply:
I will change it to is_file_to_skip.